deploy/helm/sumologic/conf/opentelemetry-operator/instrumentation.cr.yaml

```diff
@@ -0,0 +1,72 @@
+{{- $ctx := . -}}
+{{ $watchNamespace := index .Values "opentelemetry-operator" "manager" "env" }}
+{{- if eq ( get $watchNamespace "WATCH_NAMESPACE" ) "" }}
+{{ fail "It is mandatory to set value for \"opentelemetry-operator.manager.env.WATCH_NAMESPACE\" when opentelemetry-operator is enabled. Value is comma separated namespaces e.g. \"ns1\\,ns2\"" }}
```

Comment:
Why is this mandatory, actually? A user may not want to use Instrumentation at all and just have the operator manage OT in all namespaces.

Reply:
If `WATCH_NAMESPACE` is not provided then in theory Operator should watch for all namespaces but it doesn't. At least in case of `Instrumentation`. 

Logs say that `Instrumentation` resource exists and became `default`
```
{"level":"info","ts":1666018961.0456557,"logger":"instrumentation-resource","msg":"default","name":"collection-sumologic-ot-operator-instr"}
{"level":"info","ts":1666018961.0561447,"logger":"instrumentation-resource","msg":"validate update","name":"collection-sumologic-ot-operator-instr"}
```

 But when I add annotation to the e.g. `Deployment` I'm getting this kind of error. Doesn't matter if annotation value is `true` or `name-of-the-instrumentation-resource`.

```
{"level":"error","ts":1666018765.0062091,"msg":"failed to select an OpenTelemetry Instrumentation instance for this pod","namespace":"","name":"","error":"no OpenTelemetry Instrumentation instances available","stacktrace":"github.com/open-telemetry/opentelemetry-operator/pkg/instrumentation.(*instPodMutator).Mutate\n\t/workspace/pkg/instrumentation/podmutator.go:102\ngithub.com/open-telemetry/opentelemetry-operator/internal/webhookhandler.(*podSidecarInjector).Handle\n\t/workspace/internal/webhookhandler/webhookhandler.go:92\nsigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).Handle\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/webhook/admission/webhook.go:169\nsigs.k8s.io/controller-runtime/pkg/webhook/admission.(*Webhook).ServeHTTP\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/webhook/admission/http.go:98\ngithub.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerInFlight.func1\n\t/go/pkg/mod/github.com/prometheus/client_golang@v1.12.2/prometheus/promhttp/instrument_server.go:40\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2109\ngithub.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerCounter.func1\n\t/go/pkg/mod/github.com/prometheus/client_golang@v1.12.2/prometheus/promhttp/instrument_server.go:117\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2109\ngithub.com/prometheus/client_golang/prometheus/promhttp.InstrumentHandlerDuration.func2\n\t/go/pkg/mod/github.com/prometheus/client_golang@v1.12.2/prometheus/promhttp/instrument_server.go:84\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2109\nnet/http.(*ServeMux).ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2487\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2947\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:1991"}
```

I believe this is something to improve in the upstream.

Reply:
Hm, but what if you don't want instrumentation at all? I think we used to have a setting to disable default instrumentation.

Reply:
You're getting instrumentation out of the box. It's just a matter of creation of `Instrumentation` resource in the specific namespace.